lib/jekyll/filters/grouping_filters.rb

```diff
@@ -0,0 +1,56 @@
+module Jekyll
+  module Filters
+    module GroupingFilters
+      # Group an array of items by a property
+      #
+      # input - the inputted Enumerable
+      # property - the property
+      #
+      # Returns an array of Hashes, each looking something like this:
+      #  {"name"  => "larry"
+      #   "items" => [...] } # all the items where `property` == "larry"
+      def group_by(input, property)
+        if groupable?(input)
+          groups = input.group_by { |item| item_property(item, property).to_s }
```

Comment:
I did this to make Rubocop happy. But I'm torn about this refactoring and the main reason is this call to `item_property`. It bounds the new module to being included to the old one.

Should we keep this as is? Should we split `item_property` to its own module? Should we split `Filters` in a totally different way?